File: Atividade Avaliativa Final/Javascript/entity/time/Time.js
export class Time {
  constructor(nome, modalidade) {
    this.nome = nome;
    this.jogadores = [];
    this.modalidade = modalidade;
  }

  adicionarJogador(jogador) {
    this.jogadores.push(jogador);
  }

  removerJogador(jogador) {}

  exibirTime() {
    if (this.jogadores.length === 0) {
      return `O time ${this.nome} não possui jogadores.`;
    }

    const jogadoresInfo = this.jogadores
      .map((jogador) => jogador.exibirDados())
      .join(" || ");
    return `${this.nome} (${this.modalidade}) -> Jogadores: [${jogadoresInfo}]`;
  }
}

File: Atividade Avaliativa Final/Python/entity/time.py


from typing import List
from entity.jogador import Jogador

class Time:
    def __init__(self, nome: str, modalidade: str):
        self.nome = nome
        self.modalidade = modalidade
        self.jogadores: List[Jogador] = []

    def adicionar_jogador(self, jogador: Jogador):
        self.jogadores.append(jogador)

    def remover_jogador(self, jogador: Jogador):
        if jogador in self.jogadores:
            self.jogadores.remove(jogador)

    def exibir_time(self) -> str:
        dados = f"Time: {self.nome} ({self.modalidade})\n"
        for jogador in self.jogadores:
            dados += f"  - {jogador.exibir_dados()}\n"
        return dados.strip()